lib/id3c/cli/command/etl/redcap_det.py

```diff
@@ -89,49 +97,56 @@ def decorated(*args, db: DatabaseSession, log_output: bool, **kwargs):
-                        LOG.debug(f"Skipping incomplete or unverified REDCap DET {det.id}")
-                        mark_skipped(db, det.id, etl_id)
-                        continue
+                        redcap_record = get_redcap_record_from_det(det.document)
 
-                    redcap_record = get_redcap_record_from_det(det.document)
+                        if not redcap_record:
+                            LOG.debug(f"REDCap record is missing or invalid.  Skipping REDCap DET {det.id}")
+                            mark_skipped(db, det.id, etl_id)
+                            continue
 
-                    if not redcap_record:
-                        LOG.debug(f"REDCap record is missing.  Skipping REDCap DET {det.id}")
-                        mark_skipped(db, det.id, etl_id)
-                        continue
+                        # Only process REDCap record if all required instruments are complete
+                        incomplete_instruments = {
+                            instrument
+                                for instrument
+                                in required_instruments
+                                if not is_complete(instrument, redcap_record)
+                        }
 
-                    # Only process REDCap record if all required instruments are complete
-                    incomplete_instruments = {
-                        instrument
-                            for instrument
-                            in required_instruments
-                            if not is_complete(instrument, redcap_record)
-                    }
+                        if incomplete_instruments:
+                            LOG.debug(f"The following required instruments «{incomplete_instruments}» are not yet marked complete. " + \
+                                      f"Skipping REDCap DET {det.id}")
+                            mark_skipped(db, det.id, etl_id)
+                            continue
 
-                    if incomplete_instruments:
-                        LOG.debug(f"The following required instruments «{incomplete_instruments}» are not yet marked complete. " + \
-                                  f"Skipping REDCap DET {det.id}")
-                        mark_skipped(db, det.id, etl_id)
-                        continue
+                        bundle = routine(db = db, cache = cache, det = det, redcap_record = redcap_record)
 
-                    bundle = routine(det = det, redcap_record = redcap_record)
+                        if not bundle:
+                            mark_skipped(db, det.id, etl_id)
+                            continue
 
-                    if log_output:
-                        print(json.dumps(bundle, indent=2))
+                        if log_output:
+                            if indent:
+                                print(json.dumps(bundle, indent=indent))
+                            else:
+                                print(json.dumps(bundle))
```

Comment:
I think the default log output should be ndjson (so `indent=0`) and any reformatting should be done externally to ID3C (so dropping the `--indent` option). Programs such as `jq` make quick work of JSON manipulation from basic pretty-printing to filtering, searching, and modification.